### PR TITLE
Configure VS Code Python settings and recommended extensions

### PR DESCRIPTION
**Pull request by jtbandes**
cc @cclauss

Seems our CI has converged on bandit and flake8.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,7 +5,12 @@
 .pydevproject
 .DS_Store
 .idea
-.vscode
+
+.vscode/*
+!.vscode/extensions.json
+!.vscode/settings.json
```
Comment (amacneil) [inline .gitignore:11]: Just remove from gitignore entirely? What else would live in this directory?

**Comment by amacneil** [comment]
Suggest waiting until we run black on the whole repo and then enable formatOnSave

File: .gitignore (hunk 1 above)
Comment (jtbandes) [reply .gitignore:11]: I have a c_cpp_properties.json in there…must’ve been automatically created? I could delete it I suppose, but it’s annoying to have it constantly show as an untracked file.

**Comment by cclauss** [comment]
I would like to recommend that we use https://pre-commit.com to run our fastest lint jobs locally.

Additionally, I would like to recommend that we use https://pre-commit.ci to run those lint jobs instead of GitHub Actions.  The advantage of pre-commit.ci is that it will auto-format a pull request.

For example, see the workflow at https://github.com/kornia/kornia/pull/1210#commits-pushed-3e05ead where I make my commits and then pre-commit.ci comes in after me and automatically blackens my modifications so I don't have to.

**Comment by jtbandes** [comment]
@cclauss Now that we are using pre-commit in this repo, I updated this PR to move some of the pre-commit tool configs into pyproject.toml so they can also be read when running the tools directly.

File: .vscode/settings.json [hunk 2]
```diff
@@ -4,4 +4,7 @@
     "python.linting.flake8Enabled": true,
     "python.linting.banditEnabled": true,
     "python.linting.pylintEnabled": false,
-}
+    "[python]": {
+        "editor.formatOnSave": true
+    },
+}
```
Comment (cclauss) [inline .vscode/settings.json]: Please add one carriage return (\n).

**Review by cclauss — CHANGES_REQUESTED** [review]

**Pull request merged**